### DISCUSSION TITLE
Get the weather forecast model and the data-set time by hours.

### DISCUSSION THREAD

**Discussion by SebSwan** [open-meteo/open-meteo#364] [Ideas]
Hello Patrick,

A feature that will be very practical is to know the model used by hours. For exemple:
-When i use "best match" as parameter, i don't know the model used in the weather data
- when i use "seamless" as parameter, i can't separate data provider. GFS is used for long term, but for short & mid term is not easy to deduct the model used.

Key/value will be great

"154: "arome1.3"
155: "arpege11kms"
...
274: "GFS"
..."

and to know the dataset used to supply datas will be helpful too.
"154: "13/06/23 12 utc"
155: "13/06/23 06 utc"
...
274: "13/06/23 00 utc"
..."

What are your thoughts about this improvements ?
Thanks!

**Comment by patrick-zippenfenig** [comment]
I won't be easy to handle all these meta data. Combining multiple models is "just" filling an array once it finds NaNs. The logic is relatively simple.

Keeping track of update times will be hard as well. Right now, the database is constantly overwriting portions of data without managing all this meta data. Some of it needs to be improved to store update times for https://github.com/open-meteo/open-meteo/issues/89. Because all this data is not yet available, implementing feature like a meta data API is quite difficult.

What do you think about you specifying the models to mix? E.g. instead of "&models=best_match,arome_seamless" you can specify "&models=arome1.3~arpege11km~gfs013,arome_seamless"?

**Reply by SebSwan** [reply]
Ok, i understand, i believed it was easier to add the source of the data in the set.

your suggestion to create the mix of forecast model by choice is a good idea. Don't hesitate to ask me for a test.

Many thanks for your work!

**Comment by SebSwan** [comment]
Hello @patrick-zippenfenig ! i come back after one year :) Did you find a solution to know models used in the "seamless" and in "best match" ? i've developed a solution by calling all models and comparing each value with the parent model "seamless", it's inefficient but it works.

Did you develop a feature on a mix creation of models as mentioned? For example, Icon D2 is used in a large part of France in the "best match", for my use "arome_hd" is a lot more revelant but i still need a longterm view like the 16 days of GFS in the result.   Thanks again !

**Reply by patrick-zippenfenig** [reply]
Unfortunately, no. It actually got more complex with more weather models that may lack certain variables (soil moisture, solar radiation, precipitation probabilities etc). There had been some progress. E.g. the update times are now tracked for each weather model: https://open-meteo.com/en/docs/model-updates

**Reply by SebSwan** [reply]
Yes, i saw it! i forget to thank you for this new feature.  Anyway, your backend is awesome, speed and availability are top notch!
